libs/dal/src/repositories/notification-template/notification-template.repository.ts

```diff
@@ -46,22 +46,56 @@ export class NotificationTemplateRepository extends BaseRepository<
     const requestQuery: NotificationTemplateQuery = {
       _id: id,
       isBlueprint: true,
-      _organizationId: NotificationTemplateRepository.getBlueprintOrganizationId() as string,
+      _environmentId: NotificationTemplateRepository.getBlueprintEnvironmentId() as string,
     };
 
     const item = await this.MongooseModel.findOne(requestQuery).populate('steps.template');
 
     return this.mapEntity(item);
   }
 
+  async findAllGroupedByCategory(): Promise<{ name: string; blueprints: NotificationTemplateEntity[] }[]> {
+    if (!NotificationTemplateRepository.getBlueprintEnvironmentId()) {
+      return {} as { name: string; blueprints: NotificationTemplateEntity[] }[];
+    }
+
+    const requestQuery: NotificationTemplateQuery = {
+      isBlueprint: true,
+      _environmentId: NotificationTemplateRepository.getBlueprintEnvironmentId() as string,
+    };
+
+    const items = await this.MongooseModel.find(requestQuery)
+      .populate('steps.template')
+      .populate('notificationGroup')
+      .lean();
+
+    const groupedItems = items.reduce((acc, item) => {
+      const notificationGroupId = item._notificationGroupId;
+      const notificationGroupName = item.notificationGroup?.name;
+
+      if (!acc[notificationGroupId as unknown as string]) {
```

Comment:
the type of `notificationGroupId` is `ObjectId`, so you don't have to cast to `as unknown as string`, but you can do `notificationGroupId.toString()`